Compile failure with new ostream code

This is a new issue.

Am I missing a header file or something?

Notably, I get this error on gcc 9.2, but I don't get this error with clang 9.

```
In file included from /workspace/3rdparty/ozo/include/ozo/io/send.h:6,
                 from /workspace/3rdparty/ozo/include/ozo/ext/std/nullopt_t.h:5,
                 from /workspace/3rdparty/ozo/include/ozo/ext/std.h:12,
                 from /workspace/3rdparty/ozo/include/ozo/io/binary_query.h:3,
                 from /workspace/3rdparty/ozo/include/ozo/impl/io.h:6,
                 from /workspace/3rdparty/ozo/include/ozo/impl/async_request.h:6,
                 from /workspace/3rdparty/ozo/include/ozo/impl/async_execute.h:4,
                 from /workspace/3rdparty/ozo/include/ozo/execute.h:3,
                 from /workspace/inc/application/db/execute_pipeline.h:16,
                 from /workspace/inc/application/controller/telemetry_data_controller.h:9,
                 from /workspace/lib/controller/telemetry_data_controller.cpp:4:
/workspace/3rdparty/ozo/include/ozo/io/ostream.h: In instantiation of ‘ozo::Require<(Integral<T> && (sizeof (T) != 1)), ozo::ostream&> ozo::ostream::write(T) [with T = int; ozo::Require<(Integral<T> && (sizeof (T) != 1)), ozo::ostream&> = ozo::ostream&]’:
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:81:45:   required from ‘ozo::ostream& ozo::write(ozo::ostream&, Ts&& ...) [with Ts = {int}]’
/workspace/3rdparty/ozo/include/ozo/io/send.h:112:10:   required from ‘ozo::ostream& ozo::send_data_frame(ozo::ostream&, const OidMap&, const In&) [with OidMap = ozo::oid_map_t<>; In = application::BasicSharedString<char>]’
/workspace/3rdparty/ozo/include/ozo/io/array.h:121:66:   required from ‘ozo::detail::send_array_impl<T>::apply(ozo::ostream&, const OidMap&, const T&) [with OidMap = ozo::oid_map_t<>; T = std::vector<application::BasicSharedString<char> >]::<lambda(const auto:26&)> [with auto:26 = application::BasicSharedString<char>]’
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/include/g++-v9/bits/stl_algo.h:3876:5:   required from ‘_Funct std::for_each(_IIter, _IIter, _Funct) [with _IIter = __gnu_cxx::__normal_iterator<const application::BasicSharedString<char>*, std::vector<application::BasicSharedString<char> > >; _Funct = ozo::detail::send_array_impl<T>::apply(ozo::ostream&, const OidMap&, const T&) [with OidMap = ozo::oid_map_t<>; T = std::vector<application::BasicSharedString<char> >]::<lambda(const auto:26&)>]’
/usr/include/boost/range/algorithm/for_each.hpp:102:6:   required from ‘UnaryFunction boost::range::for_each(const SinglePassRange&, UnaryFunction) [with SinglePassRange = std::vector<application::BasicSharedString<char> >; UnaryFunction = ozo::detail::send_array_impl<T>::apply(ozo::ostream&, const OidMap&, const T&) [with OidMap = ozo::oid_map_t<>; T = std::vector<application::BasicSharedString<char> >]::<lambda(const auto:26&)>]’
/workspace/3rdparty/ozo/include/ozo/io/array.h:121:24:   [ skipping 47 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
/usr/include/boost/asio/async_result.hpp:151:49:   required from ‘static boost::asio::async_result<CompletionToken, Signature>::return_type boost::asio::async_result<CompletionToken, Signature>::initiate(Initiation&&, RawCompletionToken&&, Args&& ...) [with Initiation = ozo::detail::initiate_async_execute; RawCompletionToken = application::ExecutePipeline<PARAMS_T>::execute_impl(std::any) [with PARAMS_T = {application::BasicSharedString<char>, application::BasicSharedString<char>, std::optional<application::BasicSharedString<char> >}]::<lambda(ozo::error_code, auto:45)>; Args = {ozo::connection_provider<ozo::connection_pool<ozo::connection_info<ozo::oid_map_t<>, ozo::none_t> >&>, std::chrono::duration<long int, std::ratio<1, 1> >&, ozo::impl::query<std::basic_string_view<char, std::char_traits<char> >, std::reference_wrapper<std::vector<application::BasicSharedString<char>, std::allocator<application::BasicSharedString<char> > > >, std::reference_wrapper<std::vector<application::BasicSharedString<char>, std::allocator<application::BasicSharedString<char> > > >, std::reference_wrapper<std::vector<std::optional<application::BasicSharedString<char> >, std::allocator<std::optional<application::BasicSharedString<char> > > > > >}; CompletionToken = application::ExecutePipeline<PARAMS_T>::execute_impl(std::any) [with PARAMS_T = {application::BasicSharedString<char>, application::BasicSharedString<char>, std::optional<application::BasicSharedString<char> >}]::<lambda(ozo::error_code, auto:45)>; Signature = void(boost::system::error_code, std::shared_ptr<ozo::pooled_connection<yamail::resource_pool::handle<ozo::connection_rep<ozo::oid_map_t<>, ozo::none_t> >, boost::asio::io_context::executor_type> >); boost::asio::async_result<CompletionToken, Signature>::return_type = void]’
/usr/include/boost/asio/async_result.hpp:364:25:   required from ‘typename std::enable_if<boost::asio::detail::async_result_has_initiate_memfn<CompletionToken, Signature>::value, decltype (boost::asio::async_result<typename std::decay<_Tp>::type, Signature>::initiate(declval<Initiation&&>(), declval<CompletionToken&&>(), (declval<Args&>)()...))>::type boost::asio::async_initiate(Initiation&&, CompletionToken&, Args&& ...) [with CompletionToken = application::ExecutePipeline<PARAMS_T>::execute_impl(std::any) [with PARAMS_T = {application::BasicSharedString<char>, application::BasicSharedString<char>, std::optional<application::BasicSharedString<char> >}]::<lambda(ozo::error_code, auto:45)>; Signature = void(boost::system::error_code, std::shared_ptr<ozo::pooled_connection<yamail::resource_pool::handle<ozo::connection_rep<ozo::oid_map_t<>, ozo::none_t> >, boost::asio::io_context::executor_type> >); Initiation = ozo::detail::initiate_async_execute; Args = {ozo::connection_provider<ozo::connection_pool<ozo::connection_info<ozo::oid_map_t<>, ozo::none_t> >&>, std::chrono::duration<long int, std::ratio<1, 1> >&, ozo::impl::query<std::basic_string_view<char, std::char_traits<char> >, std::reference_wrapper<std::vector<application::BasicSharedString<char>, std::allocator<application::BasicSharedString<char> > > >, std::reference_wrapper<std::vector<application::BasicSharedString<char>, std::allocator<application::BasicSharedString<char> > > >, std::reference_wrapper<std::vector<std::optional<application::BasicSharedString<char> >, std::allocator<std::optional<application::BasicSharedString<char> > > > > >}; typename std::enable_if<boost::asio::detail::async_result_has_initiate_memfn<CompletionToken, Signature>::value, decltype (boost::asio::async_result<typename std::decay<_Tp>::type, Signature>::initiate(declval<Initiation&&>(), declval<CompletionToken&&>(), (declval<Args&>)()...))>::type = void]’
/workspace/3rdparty/ozo/include/ozo/execute.h:53:69:   required from ‘decltype(auto) ozo::execute_op<Initiator>::operator()(P&&, Q&&, TimeConstraint, CompletionToken&&) const [with P = ozo::connection_provider<ozo::connection_pool<ozo::connection_info<> >&>; Q = ozo::impl::query<std::basic_string_view<char>, std::reference_wrapper<std::vector<application::BasicSharedString<char>, std::allocator<application::BasicSharedString<char> > > >, std::reference_wrapper<std::vector<application::BasicSharedString<char>, std::allocator<application::BasicSharedString<char> > > >, std::reference_wrapper<std::vector<std::optional<application::BasicSharedString<char> >, std::allocator<std::optional<application::BasicSharedString<char> > > > > >; TimeConstraint = std::chrono::duration<long int>; CompletionToken = application::ExecutePipeline<PARAMS_T>::execute_impl(std::any) [with PARAMS_T = {application::BasicSharedString<char>, application::BasicSharedString<char>, std::optional<application::BasicSharedString<char> >}]::<lambda(ozo::error_code, auto:45)>; Initiator = ozo::detail::initiate_async_execute]’
/workspace/inc/application/db/execute_pipeline.h:221:17:   required from ‘void application::ExecutePipeline<PARAMS_T>::execute_impl(std::any) [with PARAMS_T = {application::BasicSharedString<char>, application::BasicSharedString<char>, std::optional<application::BasicSharedString<char> >}]’
/workspace/inc/application/db/execute_pipeline.h:158:5:   required from ‘void application::ExecutePipeline<PARAMS_T>::execute(std::any, ARGS_T&& ...) [with ARGS_T = {application::BasicSharedString<char>, application::BasicSharedString<char>, std::optional<application::BasicSharedString<char> >}; PARAMS_T = {application::BasicSharedString<char>, application::BasicSharedString<char>, std::optional<application::BasicSharedString<char> >}]’
/workspace/lib/controller/telemetry_data_controller.cpp:39:113:   required from here
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:51:25: error: no matching function for call to ‘ozo::ostream::write(ozo::detail::typed_buffer<int>&)’
   51 |         return write(buf);
      |                         ^
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:25:14: note: candidate: ‘ozo::ostream& ozo::ostream::write(const char_type*, std::streamsize)’
   25 |     ostream& write(const char_type* s, std::streamsize n) {
      |              ^~~~~
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:25:14: note:   candidate expects 2 arguments, 1 provided
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:36:54: note: candidate: ‘template<class T> ozo::Require<(Integral<T> && (sizeof (T) == 1)), ozo::ostream&> ozo::ostream::write(T)’
   36 |     Require<Integral<T> && sizeof(T) == 1, ostream&> write(T in) {
      |                                                      ^~~~~
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:36:54: note:   template argument deduction/substitution failed:
In file included from /usr/include/boost/hana/detail/decay.hpp:16,
                 from /usr/include/boost/hana/pair.hpp:16,
                 from /usr/include/boost/hana/detail/struct_macros.hpp:27,
                 from /usr/include/boost/hana/define_struct.hpp:15,
                 from /workspace/inc/application/controller/telemetry_config_interfaces.h:8,
                 from /workspace/inc/application/controller/telemetry_data_controller.h:7,
                 from /workspace/lib/controller/telemetry_data_controller.cpp:4:
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/include/g++-v9/type_traits: In substitution of ‘template<bool _Cond, class _Tp> using enable_if_t = typename std::enable_if::type [with bool _Cond = false; _Tp = ozo::ostream&]’:
/workspace/3rdparty/ozo/include/ozo/core/concept.h:56:7:   required by substitution of ‘template<bool Condition, class Type> using Require = std::enable_if_t<Condition, Type> [with bool Condition = (((bool)ozo::Integral<ozo::detail::typed_buffer<int> >) && (sizeof (ozo::detail::typed_buffer<int>) == 1)); Type = ozo::ostream&]’
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:36:54:   required by substitution of ‘template<class T> ozo::Require<(Integral<T> && (sizeof (T) == 1)), ozo::ostream&> ozo::ostream::write(T) [with T = ozo::detail::typed_buffer<int>]’
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:51:25:   required from ‘ozo::Require<(Integral<T> && (sizeof (T) != 1)), ozo::ostream&> ozo::ostream::write(T) [with T = int; ozo::Require<(Integral<T> && (sizeof (T) != 1)), ozo::ostream&> = ozo::ostream&]’
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:81:45:   required from ‘ozo::ostream& ozo::write(ozo::ostream&, Ts&& ...) [with Ts = {int}]’
/workspace/3rdparty/ozo/include/ozo/io/send.h:112:10:   required from ‘ozo::ostream& ozo::send_data_frame(ozo::ostream&, const OidMap&, const In&) [with OidMap = ozo::oid_map_t<>; In = application::BasicSharedString<char>]’
/workspace/3rdparty/ozo/include/ozo/io/array.h:121:66:   [ skipping 50 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
/usr/include/boost/asio/async_result.hpp:151:49:   required from ‘static boost::asio::async_result<CompletionToken, Signature>::return_type boost::asio::async_result<CompletionToken, Signature>::initiate(Initiation&&, RawCompletionToken&&, Args&& ...) [with Initiation = ozo::detail::initiate_async_execute; RawCompletionToken = application::ExecutePipeline<PARAMS_T>::execute_impl(std::any) [with PARAMS_T = {application::BasicSharedString<char>, application::BasicSharedString<char>, std::optional<application::BasicSharedString<char> >}]::<lambda(ozo::error_code, auto:45)>; Args = {ozo::connection_provider<ozo::connection_pool<ozo::connection_info<ozo::oid_map_t<>, ozo::none_t> >&>, std::chrono::duration<long int, std::ratio<1, 1> >&, ozo::impl::query<std::basic_string_view<char, std::char_traits<char> >, std::reference_wrapper<std::vector<application::BasicSharedString<char>, std::allocator<application::BasicSharedString<char> > > >, std::reference_wrapper<std::vector<application::BasicSharedString<char>, std::allocator<application::BasicSharedString<char> > > >, std::reference_wrapper<std::vector<std::optional<application::BasicSharedString<char> >, std::allocator<std::optional<application::BasicSharedString<char> > > > > >}; CompletionToken = application::ExecutePipeline<PARAMS_T>::execute_impl(std::any) [with PARAMS_T = {application::BasicSharedString<char>, application::BasicSharedString<char>, std::optional<application::BasicSharedString<char> >}]::<lambda(ozo::error_code, auto:45)>; Signature = void(boost::system::error_code, std::shared_ptr<ozo::pooled_connection<yamail::resource_pool::handle<ozo::connection_rep<ozo::oid_map_t<>, ozo::none_t> >, boost::asio::io_context::executor_type> >); boost::asio::async_result<CompletionToken, Signature>::return_type = void]’
/usr/include/boost/asio/async_result.hpp:364:25:   required from ‘typename std::enable_if<boost::asio::detail::async_result_has_initiate_memfn<CompletionToken, Signature>::value, decltype (boost::asio::async_result<typename std::decay<_Tp>::type, Signature>::initiate(declval<Initiation&&>(), declval<CompletionToken&&>(), (declval<Args&>)()...))>::type boost::asio::async_initiate(Initiation&&, CompletionToken&, Args&& ...) [with CompletionToken = application::ExecutePipeline<PARAMS_T>::execute_impl(std::any) [with PARAMS_T = {application::BasicSharedString<char>, application::BasicSharedString<char>, std::optional<application::BasicSharedString<char> >}]::<lambda(ozo::error_code, auto:45)>; Signature = void(boost::system::error_code, std::shared_ptr<ozo::pooled_connection<yamail::resource_pool::handle<ozo::connection_rep<ozo::oid_map_t<>, ozo::none_t> >, boost::asio::io_context::executor_type> >); Initiation = ozo::detail::initiate_async_execute; Args = {ozo::connection_provider<ozo::connection_pool<ozo::connection_info<ozo::oid_map_t<>, ozo::none_t> >&>, std::chrono::duration<long int, std::ratio<1, 1> >&, ozo::impl::query<std::basic_string_view<char, std::char_traits<char> >, std::reference_wrapper<std::vector<application::BasicSharedString<char>, std::allocator<application::BasicSharedString<char> > > >, std::reference_wrapper<std::vector<application::BasicSharedString<char>, std::allocator<application::BasicSharedString<char> > > >, std::reference_wrapper<std::vector<std::optional<application::BasicSharedString<char> >, std::allocator<std::optional<application::BasicSharedString<char> > > > > >}; typename std::enable_if<boost::asio::detail::async_result_has_initiate_memfn<CompletionToken, Signature>::value, decltype (boost::asio::async_result<typename std::decay<_Tp>::type, Signature>::initiate(declval<Initiation&&>(), declval<CompletionToken&&>(), (declval<Args&>)()...))>::type = void]’
/workspace/3rdparty/ozo/include/ozo/execute.h:53:69:   required from ‘decltype(auto) ozo::execute_op<Initiator>::operator()(P&&, Q&&, TimeConstraint, CompletionToken&&) const [with P = ozo::connection_provider<ozo::connection_pool<ozo::connection_info<> >&>; Q = ozo::impl::query<std::basic_string_view<char>, std::reference_wrapper<std::vector<application::BasicSharedString<char>, std::allocator<application::BasicSharedString<char> > > >, std::reference_wrapper<std::vector<application::BasicSharedString<char>, std::allocator<application::BasicSharedString<char> > > >, std::reference_wrapper<std::vector<std::optional<application::BasicSharedString<char> >, std::allocator<std::optional<application::BasicSharedString<char> > > > > >; TimeConstraint = std::chrono::duration<long int>; CompletionToken = application::ExecutePipeline<PARAMS_T>::execute_impl(std::any) [with PARAMS_T = {application::BasicSharedString<char>, application::BasicSharedString<char>, std::optional<application::BasicSharedString<char> >}]::<lambda(ozo::error_code, auto:45)>; Initiator = ozo::detail::initiate_async_execute]’
/workspace/inc/application/db/execute_pipeline.h:221:17:   required from ‘void application::ExecutePipeline<PARAMS_T>::execute_impl(std::any) [with PARAMS_T = {application::BasicSharedString<char>, application::BasicSharedString<char>, std::optional<application::BasicSharedString<char> >}]’
/workspace/inc/application/db/execute_pipeline.h:158:5:   required from ‘void application::ExecutePipeline<PARAMS_T>::execute(std::any, ARGS_T&& ...) [with ARGS_T = {application::BasicSharedString<char>, application::BasicSharedString<char>, std::optional<application::BasicSharedString<char> >}; PARAMS_T = {application::BasicSharedString<char>, application::BasicSharedString<char>, std::optional<application::BasicSharedString<char> >}]’
/workspace/lib/controller/telemetry_data_controller.cpp:39:113:   required from here
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/include/g++-v9/type_traits:2384:11: error: no type named ‘type’ in ‘struct std::enable_if<false, ozo::ostream&>’
 2384 |     using enable_if_t = typename enable_if<_Cond, _Tp>::type;
      |           ^~~~~~~~~~~
In file included from /workspace/3rdparty/ozo/include/ozo/io/send.h:6,
                 from /workspace/3rdparty/ozo/include/ozo/ext/std/nullopt_t.h:5,
                 from /workspace/3rdparty/ozo/include/ozo/ext/std.h:12,
                 from /workspace/3rdparty/ozo/include/ozo/io/binary_query.h:3,
                 from /workspace/3rdparty/ozo/include/ozo/impl/io.h:6,
                 from /workspace/3rdparty/ozo/include/ozo/impl/async_request.h:6,
                 from /workspace/3rdparty/ozo/include/ozo/impl/async_execute.h:4,
                 from /workspace/3rdparty/ozo/include/ozo/execute.h:3,
                 from /workspace/inc/application/db/execute_pipeline.h:16,
                 from /workspace/inc/application/controller/telemetry_data_controller.h:9,
                 from /workspace/lib/controller/telemetry_data_controller.cpp:4:
/workspace/3rdparty/ozo/include/ozo/io/ostream.h: In instantiation of ‘ozo::Require<(Integral<T> && (sizeof (T) != 1)), ozo::ostream&> ozo::ostream::write(T) [with T = int; ozo::Require<(Integral<T> && (sizeof (T) != 1)), ozo::ostream&> = ozo::ostream&]’:
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:81:45:   required from ‘ozo::ostream& ozo::write(ozo::ostream&, Ts&& ...) [with Ts = {int}]’
/workspace/3rdparty/ozo/include/ozo/io/send.h:112:10:   required from ‘ozo::ostream& ozo::send_data_frame(ozo::ostream&, const OidMap&, const In&) [with OidMap = ozo::oid_map_t<>; In = application::BasicSharedString<char>]’
/workspace/3rdparty/ozo/include/ozo/io/array.h:121:66:   required from ‘ozo::detail::send_array_impl<T>::apply(ozo::ostream&, const OidMap&, const T&) [with OidMap = ozo::oid_map_t<>; T = std::vector<application::BasicSharedString<char> >]::<lambda(const auto:26&)> [with auto:26 = application::BasicSharedString<char>]’
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/include/g++-v9/bits/stl_algo.h:3876:5:   required from ‘_Funct std::for_each(_IIter, _IIter, _Funct) [with _IIter = __gnu_cxx::__normal_iterator<const application::BasicSharedString<char>*, std::vector<application::BasicSharedString<char> > >; _Funct = ozo::detail::send_array_impl<T>::apply(ozo::ostream&, const OidMap&, const T&) [with OidMap = ozo::oid_map_t<>; T = std::vector<application::BasicSharedString<char> >]::<lambda(const auto:26&)>]’
/usr/include/boost/range/algorithm/for_each.hpp:102:6:   required from ‘UnaryFunction boost::range::for_each(const SinglePassRange&, UnaryFunction) [with SinglePassRange = std::vector<application::BasicSharedString<char> >; UnaryFunction = ozo::detail::send_array_impl<T>::apply(ozo::ostream&, const OidMap&, const T&) [with OidMap = ozo::oid_map_t<>; T = std::vector<application::BasicSharedString<char> >]::<lambda(const auto:26&)>]’
/workspace/3rdparty/ozo/include/ozo/io/array.h:121:24:   [ skipping 47 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
/usr/include/boost/asio/async_result.hpp:151:49:   required from ‘static boost::asio::async_result<CompletionToken, Signature>::return_type boost::asio::async_result<CompletionToken, Signature>::initiate(Initiation&&, RawCompletionToken&&, Args&& ...) [with Initiation = ozo::detail::initiate_async_execute; RawCompletionToken = application::ExecutePipeline<PARAMS_T>::execute_impl(std::any) [with PARAMS_T = {application::BasicSharedString<char>, application::BasicSharedString<char>, std::optional<application::BasicSharedString<char> >}]::<lambda(ozo::error_code, auto:45)>; Args = {ozo::connection_provider<ozo::connection_pool<ozo::connection_info<ozo::oid_map_t<>, ozo::none_t> >&>, std::chrono::duration<long int, std::ratio<1, 1> >&, ozo::impl::query<std::basic_string_view<char, std::char_traits<char> >, std::reference_wrapper<std::vector<application::BasicSharedString<char>, std::allocator<application::BasicSharedString<char> > > >, std::reference_wrapper<std::vector<application::BasicSharedString<char>, std::allocator<application::BasicSharedString<char> > > >, std::reference_wrapper<std::vector<std::optional<application::BasicSharedString<char> >, std::allocator<std::optional<application::BasicSharedString<char> > > > > >}; CompletionToken = application::ExecutePipeline<PARAMS_T>::execute_impl(std::any) [with PARAMS_T = {application::BasicSharedString<char>, application::BasicSharedString<char>, std::optional<application::BasicSharedString<char> >}]::<lambda(ozo::error_code, auto:45)>; Signature = void(boost::system::error_code, std::shared_ptr<ozo::pooled_connection<yamail::resource_pool::handle<ozo::connection_rep<ozo::oid_map_t<>, ozo::none_t> >, boost::asio::io_context::executor_type> >); boost::asio::async_result<CompletionToken, Signature>::return_type = void]’
/usr/include/boost/asio/async_result.hpp:364:25:   required from ‘typename std::enable_if<boost::asio::detail::async_result_has_initiate_memfn<CompletionToken, Signature>::value, decltype (boost::asio::async_result<typename std::decay<_Tp>::type, Signature>::initiate(declval<Initiation&&>(), declval<CompletionToken&&>(), (declval<Args&>)()...))>::type boost::asio::async_initiate(Initiation&&, CompletionToken&, Args&& ...) [with CompletionToken = application::ExecutePipeline<PARAMS_T>::execute_impl(std::any) [with PARAMS_T = {application::BasicSharedString<char>, application::BasicSharedString<char>, std::optional<application::BasicSharedString<char> >}]::<lambda(ozo::error_code, auto:45)>; Signature = void(boost::system::error_code, std::shared_ptr<ozo::pooled_connection<yamail::resource_pool::handle<ozo::connection_rep<ozo::oid_map_t<>, ozo::none_t> >, boost::asio::io_context::executor_type> >); Initiation = ozo::detail::initiate_async_execute; Args = {ozo::connection_provider<ozo::connection_pool<ozo::connection_info<ozo::oid_map_t<>, ozo::none_t> >&>, std::chrono::duration<long int, std::ratio<1, 1> >&, ozo::impl::query<std::basic_string_view<char, std::char_traits<char> >, std::reference_wrapper<std::vector<application::BasicSharedString<char>, std::allocator<application::BasicSharedString<char> > > >, std::reference_wrapper<std::vector<application::BasicSharedString<char>, std::allocator<application::BasicSharedString<char> > > >, std::reference_wrapper<std::vector<std::optional<application::BasicSharedString<char> >, std::allocator<std::optional<application::BasicSharedString<char> > > > > >}; typename std::enable_if<boost::asio::detail::async_result_has_initiate_memfn<CompletionToken, Signature>::value, decltype (boost::asio::async_result<typename std::decay<_Tp>::type, Signature>::initiate(declval<Initiation&&>(), declval<CompletionToken&&>(), (declval<Args&>)()...))>::type = void]’
/workspace/3rdparty/ozo/include/ozo/execute.h:53:69:   required from ‘decltype(auto) ozo::execute_op<Initiator>::operator()(P&&, Q&&, TimeConstraint, CompletionToken&&) const [with P = ozo::connection_provider<ozo::connection_pool<ozo::connection_info<> >&>; Q = ozo::impl::query<std::basic_string_view<char>, std::reference_wrapper<std::vector<application::BasicSharedString<char>, std::allocator<application::BasicSharedString<char> > > >, std::reference_wrapper<std::vector<application::BasicSharedString<char>, std::allocator<application::BasicSharedString<char> > > >, std::reference_wrapper<std::vector<std::optional<application::BasicSharedString<char> >, std::allocator<std::optional<application::BasicSharedString<char> > > > > >; TimeConstraint = std::chrono::duration<long int>; CompletionToken = application::ExecutePipeline<PARAMS_T>::execute_impl(std::any) [with PARAMS_T = {application::BasicSharedString<char>, application::BasicSharedString<char>, std::optional<application::BasicSharedString<char> >}]::<lambda(ozo::error_code, auto:45)>; Initiator = ozo::detail::initiate_async_execute]’
/workspace/inc/application/db/execute_pipeline.h:221:17:   required from ‘void application::ExecutePipeline<PARAMS_T>::execute_impl(std::any) [with PARAMS_T = {application::BasicSharedString<char>, application::BasicSharedString<char>, std::optional<application::BasicSharedString<char> >}]’
/workspace/inc/application/db/execute_pipeline.h:158:5:   required from ‘void application::ExecutePipeline<PARAMS_T>::execute(std::any, ARGS_T&& ...) [with ARGS_T = {application::BasicSharedString<char>, application::BasicSharedString<char>, std::optional<application::BasicSharedString<char> >}; PARAMS_T = {application::BasicSharedString<char>, application::BasicSharedString<char>, std::optional<application::BasicSharedString<char> >}]’
/workspace/lib/controller/telemetry_data_controller.cpp:39:113:   required from here
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:41:43: note: candidate: ‘template<class T> ozo::Require<RawDataReadable<T>, ozo::ostream&> ozo::ostream::write(const T&)’
   41 |     Require<RawDataReadable<T>, ostream&> write(const T& in) {
      |                                           ^~~~~
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:41:43: note:   template argument deduction/substitution failed:
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:48:54: note: candidate: ‘template<class T> ozo::Require<(Integral<T> && (sizeof (T) != 1)), ozo::ostream&> ozo::ostream::write(T)’
   48 |     Require<Integral<T> && sizeof(T) != 1, ostream&> write(T in) {
      |                                                      ^~~~~
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:48:54: note:   template argument deduction/substitution failed:
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:55:41: note: candidate: ‘template<class T> ozo::Require<FloatingPoint<T>, ozo::ostream&> ozo::ostream::write(T)’
   55 |     Require<FloatingPoint<T>, ostream&> write(T in) {
      |                                         ^~~~~
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:55:41: note:   template argument deduction/substitution failed:
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:59:14: note: candidate: ‘ozo::ostream& ozo::ostream::write(bool)’
   59 |     ostream& write(bool in) {
      |              ^~~~~
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:59:25: note:   no known conversion for argument 1 from ‘ozo::detail::typed_buffer<int>’ to ‘bool’
   59 |     ostream& write(bool in) {
      |                    ~~~~~^~
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:65:40: note: candidate: ‘template<class T> ozo::Require<HanaSequence<T>, ozo::ostream&> ozo::ostream::write(const T&)’
   65 |     Require<HanaSequence<T>, ostream&> write(const T& in) {
      |                                        ^~~~~
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:65:40: note:   template argument deduction/substitution failed:
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:71:38: note: candidate: ‘template<class T> ozo::Require<HanaStruct<T>, ozo::ostream&> ozo::ostream::write(const T&)’
   71 |     Require<HanaStruct<T>, ostream&> write(const T& in) {
      |                                      ^~~~~
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:71:38: note:   template argument deduction/substitution failed:
In file included from /workspace/3rdparty/ozo/include/ozo/io/send.h:6,
                 from /workspace/3rdparty/ozo/include/ozo/ext/std/nullopt_t.h:5,
                 from /workspace/3rdparty/ozo/include/ozo/ext/std.h:12,
                 from /workspace/3rdparty/ozo/include/ozo/io/binary_query.h:3,
                 from /workspace/3rdparty/ozo/include/ozo/impl/io.h:6,
                 from /workspace/3rdparty/ozo/include/ozo/impl/async_request.h:6,
                 from /workspace/3rdparty/ozo/include/ozo/request.h:3,
                 from /workspace/inc/application/db/request.h:7,
                 from /workspace/inc/application/db/request_pipeline.h:7,
                 from /workspace/inc/application/controller/device_configuration_view.h:10,
                 from /workspace/lib/controller/device_configuration_view.cpp:5:
/workspace/3rdparty/ozo/include/ozo/io/ostream.h: In instantiation of ‘ozo::Require<(Integral<T> && (sizeof (T) != 1)), ozo::ostream&> ozo::ostream::write(T) [with T = int; ozo::Require<(Integral<T> && (sizeof (T) != 1)), ozo::ostream&> = ozo::ostream&]’:
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:81:45:   required from ‘ozo::ostream& ozo::write(ozo::ostream&, Ts&& ...) [with Ts = {int}]’
/workspace/3rdparty/ozo/include/ozo/io/send.h:112:10:   required from ‘ozo::ostream& ozo::send_data_frame(ozo::ostream&, const OidMap&, const In&) [with OidMap = ozo::oid_map_t<>; In = application::BasicSharedString<char>]’
/workspace/3rdparty/ozo/include/ozo/io/array.h:121:66:   required from ‘ozo::detail::send_array_impl<T>::apply(ozo::ostream&, const OidMap&, const T&) [with OidMap = ozo::oid_map_t<>; T = std::vector<application::BasicSharedString<char> >]::<lambda(const auto:19&)> [with auto:19 = application::BasicSharedString<char>]’
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/include/g++-v9/bits/stl_algo.h:3876:5:   required from ‘_Funct std::for_each(_IIter, _IIter, _Funct) [with _IIter = __gnu_cxx::__normal_iterator<const application::BasicSharedString<char>*, std::vector<application::BasicSharedString<char> > >; _Funct = ozo::detail::send_array_impl<T>::apply(ozo::ostream&, const OidMap&, const T&) [with OidMap = ozo::oid_map_t<>; T = std::vector<application::BasicSharedString<char> >]::<lambda(const auto:19&)>]’
/usr/include/boost/range/algorithm/for_each.hpp:102:6:   required from ‘UnaryFunction boost::range::for_each(const SinglePassRange&, UnaryFunction) [with SinglePassRange = std::vector<application::BasicSharedString<char> >; UnaryFunction = ozo::detail::send_array_impl<T>::apply(ozo::ostream&, const OidMap&, const T&) [with OidMap = ozo::oid_map_t<>; T = std::vector<application::BasicSharedString<char> >]::<lambda(const auto:19&)>]’
/workspace/3rdparty/ozo/include/ozo/io/array.h:121:24:   [ skipping 49 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
/usr/include/boost/asio/async_result.hpp:364:25:   required from ‘typename std::enable_if<boost::asio::detail::async_result_has_initiate_memfn<CompletionToken, Signature>::value, decltype (boost::asio::async_result<typename std::decay<_Tp>::type, Signature>::initiate(declval<Initiation&&>(), declval<CompletionToken&&>(), (declval<Args&>)()...))>::type boost::asio::async_initiate(Initiation&&, CompletionToken&, Args&& ...) [with CompletionToken = application::impl::RequestWrapper<std::vector<application::DeviceConfig, std::allocator<application::DeviceConfig> >, application::RequestPipeline<RETURN_T, PARAMS_T>::request_impl(std::any) [with RETURN_T = application::DeviceConfig; PARAMS_T = {application::BasicSharedString<char>}]::<lambda(ozo::error_code, auto:45, std::vector<application::DeviceConfig, std::allocator<application::DeviceConfig> >)> >; Signature = void(boost::system::error_code, std::shared_ptr<ozo::pooled_connection<yamail::resource_pool::handle<ozo::connection_rep<ozo::oid_map_t<>, ozo::none_t> >, boost::asio::io_context::executor_type> >); Initiation = ozo::detail::initiate_async_request; Args = {ozo::connection_provider<ozo::connection_pool<ozo::connection_info<ozo::oid_map_t<>, ozo::none_t> >&>, std::chrono::duration<long int, std::ratio<1, 1> >&, ozo::impl::query<std::basic_string_view<char, std::char_traits<char> >, std::reference_wrapper<std::vector<application::BasicSharedString<char>, std::allocator<application::BasicSharedString<char> > > > >, std::back_insert_iterator<std::vector<application::DeviceConfig, std::allocator<application::DeviceConfig> > >}; typename std::enable_if<boost::asio::detail::async_result_has_initiate_memfn<CompletionToken, Signature>::value, decltype (boost::asio::async_result<typename std::decay<_Tp>::type, Signature>::initiate(declval<Initiation&&>(), declval<CompletionToken&&>(), (declval<Args&>)()...))>::type = void]’
/workspace/3rdparty/ozo/include/ozo/request.h:94:69:   required from ‘decltype(auto) ozo::request_op<Initiator>::operator()(P&&, Q&&, TimeConstraint, Out, CompletionToken&&) const [with P = ozo::connection_provider<ozo::connection_pool<ozo::connection_info<> >&>; Q = ozo::impl::query<std::basic_string_view<char>, std::reference_wrapper<std::vector<application::BasicSharedString<char>, std::allocator<application::BasicSharedString<char> > > > >; TimeConstraint = std::chrono::duration<long int>; Out = std::back_insert_iterator<std::vector<application::DeviceConfig, std::allocator<application::DeviceConfig> > >; CompletionToken = application::impl::RequestWrapper<std::vector<application::DeviceConfig, std::allocator<application::DeviceConfig> >, application::RequestPipeline<RETURN_T, PARAMS_T>::request_impl(std::any) [with RETURN_T = application::DeviceConfig; PARAMS_T = {application::BasicSharedString<char>}]::<lambda(ozo::error_code, auto:45, std::vector<application::DeviceConfig, std::allocator<application::DeviceConfig> >)> >; Initiator = ozo::detail::initiate_async_request]’
/workspace/inc/application/db/request.h:87:24:   required from ‘decltype(auto) application::request(CONNECTION_T&&, QUERY_T&&, TIMEOUT_T&&, HANDLER_T&&) [with RETURN_T = std::vector<application::DeviceConfig, std::allocator<application::DeviceConfig> >; CONNECTION_T = ozo::connection_provider<ozo::connection_pool<ozo::connection_info<> >&>; QUERY_T = ozo::impl::query<std::basic_string_view<char>, std::reference_wrapper<std::vector<application::BasicSharedString<char>, std::allocator<application::BasicSharedString<char> > > > >; TIMEOUT_T = std::chrono::duration<long int>; HANDLER_T = application::RequestPipeline<RETURN_T, PARAMS_T>::request_impl(std::any) [with RETURN_T = application::DeviceConfig; PARAMS_T = {application::BasicSharedString<char>}]::<lambda(ozo::error_code, auto:45, std::vector<application::DeviceConfig, std::allocator<application::DeviceConfig> >)>]’
/workspace/inc/application/db/request_pipeline.h:248:47:   required from ‘void application::RequestPipeline<RETURN_T, PARAMS_T>::request_impl(std::any) [with RETURN_T = application::DeviceConfig; PARAMS_T = {application::BasicSharedString<char>}]’
/workspace/inc/application/db/request_pipeline.h:166:5:   required from ‘void application::RequestPipeline<RETURN_T, PARAMS_T>::request(std::any, application::RequestPipeline<RETURN_T, PARAMS_T>::Callback_t, ARGS_T&& ...) [with ARGS_T = {application::BasicSharedString<char>}; RETURN_T = application::DeviceConfig; PARAMS_T = {application::BasicSharedString<char>}; application::RequestPipeline<RETURN_T, PARAMS_T>::Callback_t = std::function<void(std::error_code, application::DeviceConfig)>]’
/workspace/lib/controller/device_configuration_view.cpp:24:105:   required from here
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:51:25: error: no matching function for call to ‘ozo::ostream::write(ozo::detail::typed_buffer<int>&)’
   51 |         return write(buf);
      |                         ^
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:25:14: note: candidate: ‘ozo::ostream& ozo::ostream::write(const char_type*, std::streamsize)’
   25 |     ostream& write(const char_type* s, std::streamsize n) {
      |              ^~~~~
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:25:14: note:   candidate expects 2 arguments, 1 provided
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:36:54: note: candidate: ‘template<class T> ozo::Require<(Integral<T> && (sizeof (T) == 1)), ozo::ostream&> ozo::ostream::write(T)’
   36 |     Require<Integral<T> && sizeof(T) == 1, ostream&> write(T in) {
      |                                                      ^~~~~
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:36:54: note:   template argument deduction/substitution failed:
In file included from /usr/include/boost/hana/detail/decay.hpp:16,
                 from /usr/include/boost/hana/pair.hpp:16,
                 from /usr/include/boost/hana/detail/struct_macros.hpp:27,
                 from /usr/include/boost/hana/define_struct.hpp:15,
                 from /workspace/inc/application/controller/telemetry_config_interfaces.h:8,
                 from /workspace/inc/application/controller/device_configuration_view.h:7,
                 from /workspace/lib/controller/device_configuration_view.cpp:5:
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/include/g++-v9/type_traits: In substitution of ‘template<bool _Cond, class _Tp> using enable_if_t = typename std::enable_if::type [with bool _Cond = false; _Tp = ozo::ostream&]’:
/workspace/3rdparty/ozo/include/ozo/core/concept.h:56:7:   required by substitution of ‘template<bool Condition, class Type> using Require = std::enable_if_t<Condition, Type> [with bool Condition = (((bool)ozo::Integral<ozo::detail::typed_buffer<int> >) && (sizeof (ozo::detail::typed_buffer<int>) == 1)); Type = ozo::ostream&]’
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:36:54:   required by substitution of ‘template<class T> ozo::Require<(Integral<T> && (sizeof (T) == 1)), ozo::ostream&> ozo::ostream::write(T) [with T = ozo::detail::typed_buffer<int>]’
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:51:25:   required from ‘ozo::Require<(Integral<T> && (sizeof (T) != 1)), ozo::ostream&> ozo::ostream::write(T) [with T = int; ozo::Require<(Integral<T> && (sizeof (T) != 1)), ozo::ostream&> = ozo::ostream&]’
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:81:45:   required from ‘ozo::ostream& ozo::write(ozo::ostream&, Ts&& ...) [with Ts = {int}]’
/workspace/3rdparty/ozo/include/ozo/io/send.h:112:10:   required from ‘ozo::ostream& ozo::send_data_frame(ozo::ostream&, const OidMap&, const In&) [with OidMap = ozo::oid_map_t<>; In = application::BasicSharedString<char>]’
/workspace/3rdparty/ozo/include/ozo/io/array.h:121:66:   [ skipping 52 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
/usr/include/boost/asio/async_result.hpp:364:25:   required from ‘typename std::enable_if<boost::asio::detail::async_result_has_initiate_memfn<CompletionToken, Signature>::value, decltype (boost::asio::async_result<typename std::decay<_Tp>::type, Signature>::initiate(declval<Initiation&&>(), declval<CompletionToken&&>(), (declval<Args&>)()...))>::type boost::asio::async_initiate(Initiation&&, CompletionToken&, Args&& ...) [with CompletionToken = application::impl::RequestWrapper<std::vector<application::DeviceConfig, std::allocator<application::DeviceConfig> >, application::RequestPipeline<RETURN_T, PARAMS_T>::request_impl(std::any) [with RETURN_T = application::DeviceConfig; PARAMS_T = {application::BasicSharedString<char>}]::<lambda(ozo::error_code, auto:45, std::vector<application::DeviceConfig, std::allocator<application::DeviceConfig> >)> >; Signature = void(boost::system::error_code, std::shared_ptr<ozo::pooled_connection<yamail::resource_pool::handle<ozo::connection_rep<ozo::oid_map_t<>, ozo::none_t> >, boost::asio::io_context::executor_type> >); Initiation = ozo::detail::initiate_async_request; Args = {ozo::connection_provider<ozo::connection_pool<ozo::connection_info<ozo::oid_map_t<>, ozo::none_t> >&>, std::chrono::duration<long int, std::ratio<1, 1> >&, ozo::impl::query<std::basic_string_view<char, std::char_traits<char> >, std::reference_wrapper<std::vector<application::BasicSharedString<char>, std::allocator<application::BasicSharedString<char> > > > >, std::back_insert_iterator<std::vector<application::DeviceConfig, std::allocator<application::DeviceConfig> > >}; typename std::enable_if<boost::asio::detail::async_result_has_initiate_memfn<CompletionToken, Signature>::value, decltype (boost::asio::async_result<typename std::decay<_Tp>::type, Signature>::initiate(declval<Initiation&&>(), declval<CompletionToken&&>(), (declval<Args&>)()...))>::type = void]’
/workspace/3rdparty/ozo/include/ozo/request.h:94:69:   required from ‘decltype(auto) ozo::request_op<Initiator>::operator()(P&&, Q&&, TimeConstraint, Out, CompletionToken&&) const [with P = ozo::connection_provider<ozo::connection_pool<ozo::connection_info<> >&>; Q = ozo::impl::query<std::basic_string_view<char>, std::reference_wrapper<std::vector<application::BasicSharedString<char>, std::allocator<application::BasicSharedString<char> > > > >; TimeConstraint = std::chrono::duration<long int>; Out = std::back_insert_iterator<std::vector<application::DeviceConfig, std::allocator<application::DeviceConfig> > >; CompletionToken = application::impl::RequestWrapper<std::vector<application::DeviceConfig, std::allocator<application::DeviceConfig> >, application::RequestPipeline<RETURN_T, PARAMS_T>::request_impl(std::any) [with RETURN_T = application::DeviceConfig; PARAMS_T = {application::BasicSharedString<char>}]::<lambda(ozo::error_code, auto:45, std::vector<application::DeviceConfig, std::allocator<application::DeviceConfig> >)> >; Initiator = ozo::detail::initiate_async_request]’
/workspace/inc/application/db/request.h:87:24:   required from ‘decltype(auto) application::request(CONNECTION_T&&, QUERY_T&&, TIMEOUT_T&&, HANDLER_T&&) [with RETURN_T = std::vector<application::DeviceConfig, std::allocator<application::DeviceConfig> >; CONNECTION_T = ozo::connection_provider<ozo::connection_pool<ozo::connection_info<> >&>; QUERY_T = ozo::impl::query<std::basic_string_view<char>, std::reference_wrapper<std::vector<application::BasicSharedString<char>, std::allocator<application::BasicSharedString<char> > > > >; TIMEOUT_T = std::chrono::duration<long int>; HANDLER_T = application::RequestPipeline<RETURN_T, PARAMS_T>::request_impl(std::any) [with RETURN_T = application::DeviceConfig; PARAMS_T = {application::BasicSharedString<char>}]::<lambda(ozo::error_code, auto:45, std::vector<application::DeviceConfig, std::allocator<application::DeviceConfig> >)>]’
/workspace/inc/application/db/request_pipeline.h:248:47:   required from ‘void application::RequestPipeline<RETURN_T, PARAMS_T>::request_impl(std::any) [with RETURN_T = application::DeviceConfig; PARAMS_T = {application::BasicSharedString<char>}]’
/workspace/inc/application/db/request_pipeline.h:166:5:   required from ‘void application::RequestPipeline<RETURN_T, PARAMS_T>::request(std::any, application::RequestPipeline<RETURN_T, PARAMS_T>::Callback_t, ARGS_T&& ...) [with ARGS_T = {application::BasicSharedString<char>}; RETURN_T = application::DeviceConfig; PARAMS_T = {application::BasicSharedString<char>}; application::RequestPipeline<RETURN_T, PARAMS_T>::Callback_t = std::function<void(std::error_code, application::DeviceConfig)>]’
/workspace/lib/controller/device_configuration_view.cpp:24:105:   required from here
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/include/g++-v9/type_traits:2384:11: error: no type named ‘type’ in ‘struct std::enable_if<false, ozo::ostream&>’
 2384 |     using enable_if_t = typename enable_if<_Cond, _Tp>::type;
      |           ^~~~~~~~~~~
In file included from /workspace/3rdparty/ozo/include/ozo/io/send.h:6,
                 from /workspace/3rdparty/ozo/include/ozo/ext/std/nullopt_t.h:5,
                 from /workspace/3rdparty/ozo/include/ozo/ext/std.h:12,
                 from /workspace/3rdparty/ozo/include/ozo/io/binary_query.h:3,
                 from /workspace/3rdparty/ozo/include/ozo/impl/io.h:6,
                 from /workspace/3rdparty/ozo/include/ozo/impl/async_request.h:6,
                 from /workspace/3rdparty/ozo/include/ozo/request.h:3,
                 from /workspace/inc/application/db/request.h:7,
                 from /workspace/inc/application/db/request_pipeline.h:7,
                 from /workspace/inc/application/controller/device_configuration_view.h:10,
                 from /workspace/lib/controller/device_configuration_view.cpp:5:
/workspace/3rdparty/ozo/include/ozo/io/ostream.h: In instantiation of ‘ozo::Require<(Integral<T> && (sizeof (T) != 1)), ozo::ostream&> ozo::ostream::write(T) [with T = int; ozo::Require<(Integral<T> && (sizeof (T) != 1)), ozo::ostream&> = ozo::ostream&]’:
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:81:45:   required from ‘ozo::ostream& ozo::write(ozo::ostream&, Ts&& ...) [with Ts = {int}]’
/workspace/3rdparty/ozo/include/ozo/io/send.h:112:10:   required from ‘ozo::ostream& ozo::send_data_frame(ozo::ostream&, const OidMap&, const In&) [with OidMap = ozo::oid_map_t<>; In = application::BasicSharedString<char>]’
/workspace/3rdparty/ozo/include/ozo/io/array.h:121:66:   required from ‘ozo::detail::send_array_impl<T>::apply(ozo::ostream&, const OidMap&, const T&) [with OidMap = ozo::oid_map_t<>; T = std::vector<application::BasicSharedString<char> >]::<lambda(const auto:19&)> [with auto:19 = application::BasicSharedString<char>]’
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/include/g++-v9/bits/stl_algo.h:3876:5:   required from ‘_Funct std::for_each(_IIter, _IIter, _Funct) [with _IIter = __gnu_cxx::__normal_iterator<const application::BasicSharedString<char>*, std::vector<application::BasicSharedString<char> > >; _Funct = ozo::detail::send_array_impl<T>::apply(ozo::ostream&, const OidMap&, const T&) [with OidMap = ozo::oid_map_t<>; T = std::vector<application::BasicSharedString<char> >]::<lambda(const auto:19&)>]’
/usr/include/boost/range/algorithm/for_each.hpp:102:6:   required from ‘UnaryFunction boost::range::for_each(const SinglePassRange&, UnaryFunction) [with SinglePassRange = std::vector<application::BasicSharedString<char> >; UnaryFunction = ozo::detail::send_array_impl<T>::apply(ozo::ostream&, const OidMap&, const T&) [with OidMap = ozo::oid_map_t<>; T = std::vector<application::BasicSharedString<char> >]::<lambda(const auto:19&)>]’
/workspace/3rdparty/ozo/include/ozo/io/array.h:121:24:   [ skipping 49 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
/usr/include/boost/asio/async_result.hpp:364:25:   required from ‘typename std::enable_if<boost::asio::detail::async_result_has_initiate_memfn<CompletionToken, Signature>::value, decltype (boost::asio::async_result<typename std::decay<_Tp>::type, Signature>::initiate(declval<Initiation&&>(), declval<CompletionToken&&>(), (declval<Args&>)()...))>::type boost::asio::async_initiate(Initiation&&, CompletionToken&, Args&& ...) [with CompletionToken = application::impl::RequestWrapper<std::vector<application::DeviceConfig, std::allocator<application::DeviceConfig> >, application::RequestPipeline<RETURN_T, PARAMS_T>::request_impl(std::any) [with RETURN_T = application::DeviceConfig; PARAMS_T = {application::BasicSharedString<char>}]::<lambda(ozo::error_code, auto:45, std::vector<application::DeviceConfig, std::allocator<application::DeviceConfig> >)> >; Signature = void(boost::system::error_code, std::shared_ptr<ozo::pooled_connection<yamail::resource_pool::handle<ozo::connection_rep<ozo::oid_map_t<>, ozo::none_t> >, boost::asio::io_context::executor_type> >); Initiation = ozo::detail::initiate_async_request; Args = {ozo::connection_provider<ozo::connection_pool<ozo::connection_info<ozo::oid_map_t<>, ozo::none_t> >&>, std::chrono::duration<long int, std::ratio<1, 1> >&, ozo::impl::query<std::basic_string_view<char, std::char_traits<char> >, std::reference_wrapper<std::vector<application::BasicSharedString<char>, std::allocator<application::BasicSharedString<char> > > > >, std::back_insert_iterator<std::vector<application::DeviceConfig, std::allocator<application::DeviceConfig> > >}; typename std::enable_if<boost::asio::detail::async_result_has_initiate_memfn<CompletionToken, Signature>::value, decltype (boost::asio::async_result<typename std::decay<_Tp>::type, Signature>::initiate(declval<Initiation&&>(), declval<CompletionToken&&>(), (declval<Args&>)()...))>::type = void]’
/workspace/3rdparty/ozo/include/ozo/request.h:94:69:   required from ‘decltype(auto) ozo::request_op<Initiator>::operator()(P&&, Q&&, TimeConstraint, Out, CompletionToken&&) const [with P = ozo::connection_provider<ozo::connection_pool<ozo::connection_info<> >&>; Q = ozo::impl::query<std::basic_string_view<char>, std::reference_wrapper<std::vector<application::BasicSharedString<char>, std::allocator<application::BasicSharedString<char> > > > >; TimeConstraint = std::chrono::duration<long int>; Out = std::back_insert_iterator<std::vector<application::DeviceConfig, std::allocator<application::DeviceConfig> > >; CompletionToken = application::impl::RequestWrapper<std::vector<application::DeviceConfig, std::allocator<application::DeviceConfig> >, application::RequestPipeline<RETURN_T, PARAMS_T>::request_impl(std::any) [with RETURN_T = application::DeviceConfig; PARAMS_T = {application::BasicSharedString<char>}]::<lambda(ozo::error_code, auto:45, std::vector<application::DeviceConfig, std::allocator<application::DeviceConfig> >)> >; Initiator = ozo::detail::initiate_async_request]’
/workspace/inc/application/db/request.h:87:24:   required from ‘decltype(auto) application::request(CONNECTION_T&&, QUERY_T&&, TIMEOUT_T&&, HANDLER_T&&) [with RETURN_T = std::vector<application::DeviceConfig, std::allocator<application::DeviceConfig> >; CONNECTION_T = ozo::connection_provider<ozo::connection_pool<ozo::connection_info<> >&>; QUERY_T = ozo::impl::query<std::basic_string_view<char>, std::reference_wrapper<std::vector<application::BasicSharedString<char>, std::allocator<application::BasicSharedString<char> > > > >; TIMEOUT_T = std::chrono::duration<long int>; HANDLER_T = application::RequestPipeline<RETURN_T, PARAMS_T>::request_impl(std::any) [with RETURN_T = application::DeviceConfig; PARAMS_T = {application::BasicSharedString<char>}]::<lambda(ozo::error_code, auto:45, std::vector<application::DeviceConfig, std::allocator<application::DeviceConfig> >)>]’
/workspace/inc/application/db/request_pipeline.h:248:47:   required from ‘void application::RequestPipeline<RETURN_T, PARAMS_T>::request_impl(std::any) [with RETURN_T = application::DeviceConfig; PARAMS_T = {application::BasicSharedString<char>}]’
/workspace/inc/application/db/request_pipeline.h:166:5:   required from ‘void application::RequestPipeline<RETURN_T, PARAMS_T>::request(std::any, application::RequestPipeline<RETURN_T, PARAMS_T>::Callback_t, ARGS_T&& ...) [with ARGS_T = {application::BasicSharedString<char>}; RETURN_T = application::DeviceConfig; PARAMS_T = {application::BasicSharedString<char>}; application::RequestPipeline<RETURN_T, PARAMS_T>::Callback_t = std::function<void(std::error_code, application::DeviceConfig)>]’
/workspace/lib/controller/device_configuration_view.cpp:24:105:   required from here
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:41:43: note: candidate: ‘template<class T> ozo::Require<RawDataReadable<T>, ozo::ostream&> ozo::ostream::write(const T&)’
   41 |     Require<RawDataReadable<T>, ostream&> write(const T& in) {
      |                                           ^~~~~
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:41:43: note:   template argument deduction/substitution failed:
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:48:54: note: candidate: ‘template<class T> ozo::Require<(Integral<T> && (sizeof (T) != 1)), ozo::ostream&> ozo::ostream::write(T)’
   48 |     Require<Integral<T> && sizeof(T) != 1, ostream&> write(T in) {
      |                                                      ^~~~~
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:48:54: note:   template argument deduction/substitution failed:
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:55:41: note: candidate: ‘template<class T> ozo::Require<FloatingPoint<T>, ozo::ostream&> ozo::ostream::write(T)’
   55 |     Require<FloatingPoint<T>, ostream&> write(T in) {
      |                                         ^~~~~
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:55:41: note:   template argument deduction/substitution failed:
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:59:14: note: candidate: ‘ozo::ostream& ozo::ostream::write(bool)’
   59 |     ostream& write(bool in) {
      |              ^~~~~
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:59:25: note:   no known conversion for argument 1 from ‘ozo::detail::typed_buffer<int>’ to ‘bool’
   59 |     ostream& write(bool in) {
      |                    ~~~~~^~
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:65:40: note: candidate: ‘template<class T> ozo::Require<HanaSequence<T>, ozo::ostream&> ozo::ostream::write(const T&)’
   65 |     Require<HanaSequence<T>, ostream&> write(const T& in) {
      |                                        ^~~~~
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:65:40: note:   template argument deduction/substitution failed:
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:71:38: note: candidate: ‘template<class T> ozo::Require<HanaStruct<T>, ozo::ostream&> ozo::ostream::write(const T&)’
   71 |     Require<HanaStruct<T>, ostream&> write(const T& in) {
      |                                      ^~~~~
/workspace/3rdparty/ozo/include/ozo/io/ostream.h:71:38: note:   template argument deduction/substitution failed:
gmake[2]: *** [lib/controller/CMakeFiles/controller.dir/build.make:89: lib/controller/CMakeFiles/controller.dir/telemetry_data_controller.cpp.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
gmake[2]: *** [lib/controller/CMakeFiles/controller.dir/build.make:76: lib/controller/CMakeFiles/controller.dir/device_configuration_view.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:1089: lib/controller/CMakeFiles/controller.dir/all] Error 2
gmake: *** [Makefile:130: all] Error 2

```

--- a/include/ozo/detail/typed_buffer.h
+++ b/include/ozo/detail/typed_buffer.h
@@ -2,25 +2,14 @@
 
 #include <ozo/core/concept.h>
 
-namespace ozo {
-namespace detail {
+namespace ozo::detail {
 
 template <typename T>
 union typed_buffer {
-    constexpr static auto size = sizeof(T);
+    static_assert(std::is_fundamental_v<T>, "T should be fundamental type");
+
     T typed;
-    char raw[size];
+    char raw[sizeof(T)];
 };
 
-template <typename T>
-constexpr char* data(typed_buffer<T>& buf) noexcept { return buf.raw;}
-
-template <typename T>
-constexpr const char* data(const typed_buffer<T>& buf) noexcept { return buf.raw;}
-
-template <typename T>
-constexpr auto size(const typed_buffer<T>& buf) noexcept { return buf.size;}
-} // namespace detail
-
-} // namespace ozo
-
+} // namespace ozo::detail

--- a/include/ozo/io/istream.h
+++ b/include/ozo/io/istream.h
@@ -69,7 +69,7 @@ public:
     template <typename T>
     Require<Integral<T> && sizeof(T) != 1, istream&> read(T& out) noexcept {
         detail::typed_buffer<T> buf;
-        read(buf);
+        read(buf.raw);
         out = detail::convert_from_big_endian(buf.typed);
         return *this;
     }

--- a/include/ozo/io/ostream.h
+++ b/include/ozo/io/ostream.h
@@ -48,7 +48,7 @@ public:
     Require<Integral<T> && sizeof(T) != 1, ostream&> write(T in) {
         detail::typed_buffer<T> buf;
         buf.typed = detail::convert_to_big_endian(in);
-        return write(buf);
+        return write(buf.raw);
     }
 
     template <typename T>

